### PR TITLE
Adding PSF content and structure

### LINKED ISSUE #7: Add content on PSF modelling methods
- [ ] Add content on the different PSF modelling methods with entasis on non-parametric methods.

- [ ] Decide on how should this information should be structured.

### PR DESCRIPTION
**Pull request by tobias-liaudat**
I'm adding a possible structure for the PSF chapter of the book (Issue #7 )

Also I'm adding possible content to the sections that we can discuss on Friday's meeting.

Note: Still unfinished work.

### CODE REVIEW

**Comment by sfarrens** [comment]
For diffraction effects perhaps an interactive function to plot an Airy disk (_e.g._ using[ Astropy function](https://docs.astropy.org/en/stable/api/astropy.convolution.AiryDisk2DKernel.html)) based on telescope aperture size and/or wavelength.

**Comment by tobias-liaudat** [comment]
Possibly useful function:

https://docs.astropy.org/en/stable/api/astropy.convolution.AiryDisk2DKernel.html

File: shearbook/psf/psf-origins.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Describe the contributors of the PSF:
+
+## Optic level:
+- Diffraction effects
+- Imperfect optics
+- Chromaticity
+  - Optical system Chromaticity
+  - Observed object SED
+
```
Comment (aboucaud) [inline shearbook/psf/psf-origins.md]: To match the amount of details given at the "detector level", I would list here : 
- mirror 
  - size and inclination
  - polishing effects
- optical system 
  - internal reflections
  - misalignment
  - wavelength dependency
- filters
  - edges
  - bandpass 
  - wavelength dependency
- number and position of the arms sustaining the camera

Concerning `observed object SED`, it feels a bit off here.  I would rather put it in a `observation conditions` subsection, along with `atmosphere (ground-based telescopes)`, `AOCS and jitter (space telescope)`

Comment (aboucaud) [reply shearbook/psf/psf-origins.md]: Also we "could" try to make a distinction between PSF and PRF, which to my knowledge corresponds to the detector + optics response, without the observations conditions. It is therefore given as a function of wavelength.

Comment (aguinot) [reply shearbook/psf/psf-origins.md]: I also think this important. Specially for ground based survey where the atmosphere play an important role. There is an effect called something like "differential atmospheric refraction" which could bias weak lensing measurement. Not sure we have to go in much details for this but talking about the wavelength dependence seems to be the minimum.

**Comment by tobias-liaudat** [comment]
I added a new preliminary version of the sources of the PSF with some examples.

We should give a nicer format for the examples and the code, but the content should be there. I could use some help over there.

If you have comments please don't hesitate to leave them here.

**Comment by sfarrens** [comment]
Nice @tobias-liaudat we can go through this tomorrow!

**Review by aboucaud — DISMISSED** [review]
Very nice start, I guess this PR should be merged to set the structure and future PR should be opened to add more content to a given file at a time.

**Comment by aboucaud** [comment]
I'll let you discuss that tomorrow of course but since I won't be there, you have my opinion 😉 

After seeing these nice a visual plots, one thing I would consider adding (although maybe not in this intro notebook) is the Fourier decomposition of the PSF you showed, since convolutions are usually done in Fourier.

**Comment by tobias-liaudat** [comment]
@aboucaud yes, it is not finished work. I'll wait for tomorrow to merge.

Maybe I'll add a third column to the plots showing the FFT.

**Comment by aguinot** [comment]
Hey!
I just finish looking at your section and I think it looks very nice! We can discuss tomorrow how technical and how much details we want to put here. You can found some comments below. We can discuss more tomorrow.

- I would move 1.1.2 to 1.1.1.1, it would make more sense to me.
- 1.1.1.1: For the ground based PSF, I agree with what you say on the Moffat for long exposures, but I don't think we should only show a Moffat.. I would at least show a Kolmogorov model which is based on the atmospheric perturbations theory.
- 1.1.1.1: I would show the space based before the ground based.
- 1.1.2.1: Be careful when you say "mirrors are not perfectly spheric". I think this is "maladroit". Some mirrors are not spheric at all on purpose and this actually correct some aberrations. I think it would be better to say that there is some imperfections in the curvature or during the fabrication. In any case I would not refer to the geometry.
- 1.1.2.1: regarding the size, I agree with your comment but from what you say it seems that we could just increase the size freely. But the larger the mirror (mono-bloc) the harder it is to conserve its shape.. There is actually Active Optic systems that correct for the distortions due to mirror own weight. Maybe this is too technical for this book..
- 1.1.2.1: for the obscuration, you mention the "arm" but the secondary (or third, or more..) or even the camera (like CFHT, LSST,..) are in front of the primary mirror and create a large obscuration (>30%). And it will have a strong effect as well.
- 1.1.2.1: In the filter I change "limit" to "select".
- 1.1.2.2: maybe we can mention the dome turbulence which is a strong contribution to the PSF..
- 1.1.2.3: Guiding errors are not specific to space observations but it might be harder to correct for it in space (just because moving the telescope is more complicated).
- 1.1.2.3: Do you have a ref for the polarisation effect?
- 1.1.2.4: I don't know if I would mentioned all those effects since some of them are "just" artifact that are a problem for the entire analysis, not only the PSF (CDE, CTI, QE). 
- 1.1.2.4: For the under sampling you can link to the "parametric model" section which solve this problem. (Right ?)
- 1.1.2.4: For the Brighter Fatter, "Each CCD in a camera is not independent...", I agree, but don't you mean "each pixels" ?

**Comment by andrevitorelli** [comment]
Computational Fluid Dynamics simulation of GMT enclosure
https://www.youtube.com/watch?v=HU1i6JzvIzY

**Comment by thuiop** [comment]
Wow, these are a lot of figures !
Some criticism : 
- The wavefront aberration figure features a lens, not a mirror, and large telescopes are made with mirrors (and you talk about mirror imperfections in the text). The second one is good.
- The HST errors one is good. It seems there is a formatting problem just after though, maybe because of indent ?
- The scattering light one is superfluous to me ; I do not think it helps much to understand that there are some reflections in the mirror. Just citing the paper if people want more details is sufficient.
- Obscurations and filters are nice.
- I do not know whether it is possible to have an animated image in a notebook but an animation of the changing atmospheric PSF would be more explicit than the Kolmogorov screens.
- The environmental one is great.
- I cannot judge for the Charge Diffusion error ; but there is a typo in the title.

Overall great job finding all those figures !
Side note : don't forget to hide the plotting function at the beginning in the notebook ; you can put it in shrbk/plot.py at the root of the repo and only import it in the notebook.

**Comment by tobias-liaudat** [comment]
@thuiop Thanks for the comments!

The wavefront aberration figure features a lens, not a mirror, and large telescopes are made with mirrors (and you talk about mirror imperfections in the text). The second one is good.
**Yeah, indeed the illustration is of a lens. I wanted to illustrate what optical path differences were but didn't found an example with mirrors :/  I think I'll keep the illustration but specifying more clearly the point you raised.** 

The HST errors one is good. It seems there is a formatting problem just after though, maybe because of indent ?
**I don't see the formatting when I build the notebook. Can you show me what does it look like?** 

The scattering light one is superfluous to me ; I do not think it helps much to understand that there are some reflections in the mirror. Just citing the paper if people want more details is sufficient.
**Yeah, it's not super illustrative. I'll ask the question in today's meeting and maybe erase it.** 

Obscurations and filters are nice.
**Thanks :)**

I do not know whether it is possible to have an animated image in a notebook but an animation of the changing atmospheric PSF would be more explicit than the Kolmogorov screens.
**Indeed, if that's a possibility, it would be better! Maybe someone else in the group can answer that question..**

The environmental one is great.
I cannot judge for the Charge Diffusion error ; but there is a typo in the title.
**Thanks, I'll correct it**

**Comment by thuiop** [comment]
It seems the formatting problem was in the notebook and it does not reflect in the book, so nevermind about that.

**Review by aguinot — DISMISSED** [review]

**Review by sfarrens — APPROVED** [review]

**Pull request merged**